plugins/modules/vm_snapshot_attach_disk.py

```diff
@@ -0,0 +1,209 @@
+"""
+
+RETURN = r"""
+records:
+  description:
+    - A list of VM Snapshot records.
+  returned: success
+  type: list
+  sample:
+    - automated_trigger_timestamp: 0
+      block_count_diff_from_serial_number: 2
+      label: snap-2
+      local_retain_until_timestamp: 0
+      remote_retain_until_timestamp: 0
+      replication: true
+      snapshot_uuid: 28d6ff95-2c31-4a1a-b3d9-47535164d6de
+      timestamp: 1679397326
+      type: USER
+      vm:
+        name: snapshot-test-vm-1
+        snapshot_serial_number: 3
+        uuid: 5e50977c-14ce-450c-8a1a-bf5c0afbcf43
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import errors, arguments
+from ..module_utils.client import Client
+from ..module_utils.rest_client import RestClient
+from ..module_utils.vm_snapshot import VMSnapshot
+from ..module_utils.typed_classes import TypedVMSnapshotToAnsible
+from typing import List, Optional
+
+
+# ++++++++++++
+# Must be reviewed - not sure if that's how this should work
+# ++++++++++++
+def attach_disk(module: AnsibleModule, rest_client: RestClient):
+    vm_name = module.params["vm_name"]  # maybe this parameter is not needed? We can get vm_uuid through the specified snapshot...
+    source_snapshot_uuid = module.params["source_snapshot_uuid"]
+    # snapshot_type = module.params["snapshot_type"]
+    source_disk_slot = module.params[
+        "source_disk_slot"
+    ]  # the higher the index, the newer the disk
+
+    vm_snapshot = VMSnapshot.get_snapshots_by_query(
+        {"uuid": source_snapshot_uuid}, rest_client
+    )[0]  # there is only 1 snapshot with the specified uuid.
+    vm_uuid = vm_snapshot["vm"]["uuid"]
+
+    # TODO: add "device_snapshots" to module_utils/vm_snapshot.py
+    #  add "device_snapshots" dict to the vm_snapshot from_hypercore() method
+    #  add "vm.block_devices" dict to the vm_snapshot from_hypercore() method
+    source_disk_uuid = vm_snapshot["device_snapshots"][source_disk_slot]
+
+    # TODO: implement this method!
+    #  should return a dict (NOT list)
+    source_disk_info = VMSnapshot.get_vm_disk_info(source_disk_uuid)
+
+    # build a payload according to /rest/v1/VirDomainBlockDevice/{uuid}/clone documentation
+    payload = dict(
+        options=dict(
+            regenerateDiskID=True,
+            readOnly=source_disk_info["read_only"],
+        ),
+        snapUUID=source_snapshot_uuid,
+        template=dict(
+            virDomainUUID=vm_uuid,
+            type=source_disk_info["type"],
+            chacheMode=source_disk_info["cache_mode"],
+            capacity=source_disk_info["capacity"],
+            shareUUID=source_disk_info["share_uuid"],
+            path=source_disk_info["path"],
+            slot=source_disk_info["slot"],
+            name=source_disk_info["name"],
+            disableSnapshotting=source_disk_info["disable_snapshotting"],
+            tieringPriorityFactor=source_disk_info["tiering_priority_factor"],
+        ),
+    )
```

Comment:
I think params `capacity`, `shareUUID`, `path`, `name` are read-only. Try to remove them from POST payload to see if you can omit them. Or check in UI what payload is sent to server for this action.

`disableSnapshotting` - I would try to omit that one too.

`type` - I think user should be able to select destination type. If I need to attach virtio disk, but my windows VM does not have virtio drivers, then type needs to be changed.